airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/upgrade_cdk/pipeline.py

```diff
@@ -55,17 +66,67 @@ async def _run(self) -> StepResult:
+        build_gradle_content = await build_gradle.contents()
+
+        old_cdk_version_required = re.search(r"cdkVersionRequired *= *'(?P<version>[0-9]*\.[0-9]*\.[0-9]*)?'", build_gradle_content)
+        # If there is no airbyte-cdk dependency, add the version
+        if old_cdk_version_required is None:
+            raise ValueError("Could not find airbyte-cdk dependency in build.gradle")
+
+        if self.new_version == "latest":
+            version_file_content = await self.context.get_repo_file(
+                "airbyte-cdk/java/airbyte-cdk/core/src/main/resources/version.properties"
+            ).contents()
+            new_version = re.search(r"version *= *(?P<version>[0-9]*\.[0-9]*\.[0-9]*)", version_file_content).group("version")
+        else:
+            new_version = self.new_version
+
+        updated_build_gradle = build_gradle_content.replace(old_cdk_version_required.group("version"), new_version)
+
+        use_local_cdk = re.search(r"useLocalCdk *=.*", updated_build_gradle)
+        if use_local_cdk is not None:
+            updated_build_gradle = updated_build_gradle.replace(use_local_cdk.group(), "useLocalCdk = false")
+
+        return og_connector_dir.with_new_file("build.gradle", updated_build_gradle)
+
+    async def upgrade_cdk_version_for_python_connector(self, og_connector_dir: Directory) -> Directory:
+        context = self.context
+        og_connector_dir = await context.get_connector_dir()
+        if "setup.py" not in await og_connector_dir.entries():
+            self.skip_reason = f"Python connector {self.context.connector.technical_name} does not have a setup.py file."
+            return None
+        setup_py = og_connector_dir.file("setup.py")
+        setup_py_content = await setup_py.contents()
+
         airbyte_cdk_dependency = re.search(
-            r"airbyte-cdk(?P<extra>\[[a-zA-Z0-9-]*\])?(?P<version>[<>=!~]+[0-9]*\.[0-9]*\.[0-9]*)?", og_setup_py_content
+            r"airbyte-cdk(?P<extra>\[[a-zA-Z0-9-]*\])?(?P<version>[<>=!~]+[0-9]*(?:\.[0-9]*)?(?:\.[0-9]*)?)?", setup_py_content
         )
         # If there is no airbyte-cdk dependency, add the version
-        if airbyte_cdk_dependency is not None:
-            new_version = f"airbyte-cdk{airbyte_cdk_dependency.group('extra') or ''}>={self.new_version}"
-            return og_setup_py_content.replace(airbyte_cdk_dependency.group(), new_version)
-        else:
+        if airbyte_cdk_dependency is None:
             raise ValueError("Could not find airbyte-cdk dependency in setup.py")
 
+        if self.new_version == "latest":
+            """
+            Get the latest version of airbyte-cdk from pypi
+            """
+            cdk_pypi_url = "https://pypi.org/pypi/airbyte-cdk/json"
+            response = requests.get(cdk_pypi_url)
+            response.raise_for_status()
+            package_info = response.json()
+            new_version = package_info["info"]["version"]
```

Comment:
Can we keep this logic in a separate function? Maybe in a new `cdk` module under `pipelines/helpers/connectors/cdk.py` 

I'd also find it useful to store your logic to get the latest Java cdk version in there.